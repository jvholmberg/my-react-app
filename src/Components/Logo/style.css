.Logo.Logo-Large {
  width: 5rem;
  height: 5rem;
}
.Logo.Logo-Medium {
  width: 3rem;
  height: 3rem;
}
.Logo.Logo-Small {
  width: 2rem;
  height: 2rem;
}
