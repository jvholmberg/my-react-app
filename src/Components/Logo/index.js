import React from 'react';
import { CraftClassName } from '../';

import './style.css';

const Logo = ({
  icon,
  large,
  medium,
  small,
  xsmall,
  xxsmall,
  className,
  ...props
}) => {
  let cn = CraftClassName({ large, medium, small, xsmall, xxsmall }, 'Logo', 'Small', 'Logo-');
  if (icon) cn = `${cn} ${icon}`;
  if (className) cn = `${cn} ${className}`;
  console.log(cn);
  return (<i className={cn} {...props} />);
};
export default Logo;
