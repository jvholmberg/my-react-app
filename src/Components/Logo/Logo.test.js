import React from 'react';
import renderer from 'react-test-renderer';

import Logo from './';

describe('Logo', () => {
  it('Empty matches snapshot', () => {
    const tree = renderer
      .create(<Logo />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('With icon matches snapshot', () => {
    const tree = renderer
      .create(<Logo icon='fas fa-bars' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('With className matches snapshot', () => {
    const tree = renderer
      .create(<Logo className='className' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
