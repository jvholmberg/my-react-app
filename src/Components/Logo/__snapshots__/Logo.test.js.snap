// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Empty matches snapshot 1`] = `
<i
  className="Logo Logo-Small"
/>
`;

exports[`Logo With className matches snapshot 1`] = `
<i
  className="Logo Logo-Small className"
/>
`;

exports[`Logo With icon matches snapshot 1`] = `
<i
  className="Logo Logo-Small fas fa-bars"
/>
`;
